### PR TITLE
feat: allow functions to force non-http mode by specifying the force-non-http header in the response

### LINKED ISSUE #296: [AWS] Allow functions to force non-HTTP mode
**Is your feature request related to a problem? Please describe.**

The way that helix-universal detects that the response should be returned as-is is based on the absence of `event.requestContext` (see https://github.com/adobe/helix-universal/blob/5eb09fc6a564e81111db2616bea195348a2ec49e/src/aws-adapter.js#L87). However, this does not work properly when a function is used as an API Gateway Authorizer.

When used as an Authorizer, `event.requestContext` **is** set. But the response needs to be returned in the raw form provided by helix-universal when `nonHttp` is `true`.

**Describe the solution you'd like**

Allow functions to explicitly ask for non-HTTP mode to be used in the response by setting a header, e.g. `force-non-http: true`

**Describe alternatives you've considered**

Not using helix-universal for this use case but that would create other problems.

**Additional context**

### PR DESCRIPTION
**Pull request by justinedelson**
Please ensure your pull request adheres to the following guidelines:
- [x] make sure to link the related issues in this description
- [ ] when merging / squashing, make sure the fixed issue references are visible in the commits, for easy compilation of release notes

## Related Issues

-  #296

Thanks for contributing!

### CODE REVIEW

File: src/aws-adapter.js [hunk 1]
```diff
@@ -179,7 +179,7 @@ export function createAdapter(opts = {}) {
         await con.log.flush();
       }
 
-      if (nonHttp) {
+      if (nonHttp || response.headers.get('force-non-http') === 'true') {
```
Comment (tripodsan) [inline src/aws-adapter.js:182]: I'm not very happy with the header name, as the `non-http` isn't really related to the response. one option would be to re-evaluate `context.invocation.event.requestContext`
and deleting it in the function. but this is also a bit ugly.

another option would be to return a status.code `0` or `999` (if fetch allows it) in order to mark it as non-http.

or use a header named `x-aws-raw-response`

Comment (justinedelson) [reply src/aws-adapter.js:182]: I like `x-aws-raw-response`.

Comment (tripodsan) [reply src/aws-adapter.js]: ok...I want to have @dominique-pfister input as well...

**Review by tripodsan — APPROVED** [review]

File: src/aws-adapter.js (hunk 1 above)
Comment (dominique-pfister) [reply src/aws-adapter.js]: > I like `x-aws-raw-response`.

same here 👍 

What I don't see at the moment, is why you'd want a HTTP request to return a non-HTTP response? Isn't it enough to directly invoke the Lambda and then benefit from `nonHttp` being true?

**Review by dominique-pfister — APPROVED** [review]
It would be ideal to somehow detect that a Lambda is being called as an authorizer, which makes it mandatory to turn to `nonHttp`, but as this can't be achieved, I think this is the best workaround.

**Pull request merged**